crates/store/re_grpc_client/src/redap/mod.rs

```diff
@@ -333,84 +327,32 @@ async fn stream_catalog_async(
 
     re_log::info!("Starting to read...");
     while let Some(result) = resp.next().await {
-        let input = TransportChunk::from(result.map_err(TonicStatusError)?);
-
-        // Catalog received from the ReDap server isn't suitable for direct conversion to a Rerun Chunk:
-        // - conversion expects "data" columns to be ListArrays, hence we need to convert any individual row column data to ListArray
-        // - conversion expects the input TransportChunk to have a ChunkId so we need to add that piece of metadata
-
-        let mut fields: Vec<ArrowField> = Vec::new();
-        let mut columns: Vec<ArrowArrayRef> = Vec::new();
-        // add the (row id) control field
-        let (row_id_field, row_id_data) = input.controls().next().ok_or(
-            StreamError::ChunkError(re_chunk::ChunkError::Malformed {
-                reason: "no control field found".to_owned(),
-            }),
-        )?;
-
-        fields.push(
-            ArrowField::new(
-                RowId::name().to_string(), // need to rename to Rerun Chunk expected control field
-                row_id_field.data_type().clone(),
-                false, /* not nullable */
-            )
-            .with_metadata(TransportChunk::field_metadata_control_column()),
-        );
-        columns.push(row_id_data.clone());
+        let entity_path = EntityPath::parse_forgiving("catalog");
 
-        // next add any timeline field
-        for (field, data) in input.timelines() {
-            fields.push(field.clone());
-            columns.push(data.clone());
-        }
+        let mut record_batch = result.map_err(TonicStatusError)?;
 
-        // now add all the 'data' fields - we slice each column array into individual arrays and then convert the whole lot into a ListArray
-        for (field, data) in input.components() {
-            let data_field_inner =
-                ArrowField::new("item", field.data_type().clone(), true /* nullable */);
-
-            let data_field = ArrowField::new(
-                field.name().clone(),
-                ArrowDataType::List(Arc::new(data_field_inner.clone())),
-                false, /* not nullable */
-            )
-            .with_metadata(TransportChunk::field_metadata_data_column());
-
-            let mut sliced: Vec<ArrowArrayRef> = Vec::new();
```

Comment:
Ha! So that's where it came from. Yeah we should provide a util in `arrow_util` that shows how to do this much more cheaply.

Reply:
* https://github.com/rerun-io/rerun/pull/8998